app/models/SubmissionProcessingResult.scala

```diff
@@ -16,16 +16,21 @@
 
 package models
 
+import models.EisSubmissionResult.EisSubmissionRejected
+
 sealed trait SubmissionProcessingResult
 
 object SubmissionProcessingResult {
 
   case object SubmissionSuccess extends SubmissionProcessingResult
 
-  sealed trait SubmissionFailure        extends SubmissionProcessingResult
+  sealed trait SubmissionFailure extends SubmissionProcessingResult
+
   case object SubmissionFailureInternal extends SubmissionFailure
+
   case object SubmissionFailureExternal extends SubmissionFailure
-  case object SubmissionFailureRejected extends SubmissionFailure
+
+  case class SubmissionFailureRejected(submissionResult: EisSubmissionRejected) extends SubmissionFailure
```

Comment:
Returning `EisSubmissionRejected` breaks all encapsulation that this was mean to provide. The controller now uses this for it's implementation

Reply:
I could change EisSubmissionRejected so that it implements the SubmissionFailure trait but that would mean removing the sealed trait.